### ISSUE TITLE
regression: trait no longer implemented 

### ISSUE THREAD

**Issue by Mark-Simulacrum** [rust-lang/rust#61563]
* sized-vec-0.3.0: [start](https://crater-reports.s3.amazonaws.com/beta-1.36-1/1.35.0/reg/sized-vec-0.3.0/log.txt) v. [end](https://crater-reports.s3.amazonaws.com/beta-1.36-1/beta-2019-05-30/reg/sized-vec-0.3.0/log.txt); cc @bodil

```
---- src/lib.rs - Vec<N, A>::from (line 1087) stdout ----
error[E0277]: the trait bound `sized_vec::Vec<_, _>: std::convert::From<generic_array::GenericArray<{integer}, typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>, typenum::bit::B1>>>` is not satisfied
 --> src/lib.rs:1095:28
  |
9 | let good_vec: Vec<U3, _> = Vec::from(array);
  |                            ^^^^^^^^^ the trait `std::convert::From<generic_array::GenericArray<{integer}, typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>, typenum::bit::B1>>>` is not implemented for `sized_vec::Vec<_, _>`
  |
  = help: the following implementations were found:
            <sized_vec::Vec<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>, typenum::bit::B0>, typenum::bit::B0>, typenum::bit::B0>, typenum::bit::B0>, A> as std::convert::From<[A; 16]>>
            <sized_vec::Vec<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>, typenum::bit::B0>, typenum::bit::B0>, typenum::bit::B0>, typenum::bit::B1>, A> as std::convert::From<[A; 17]>>
            <sized_vec::Vec<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>, typenum::bit::B0>, typenum::bit::B0>, typenum::bit::B1>, typenum::bit::B0>, A> as std::convert::From<[A; 18]>>
            <sized_vec::Vec<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>, typenum::bit::B0>, typenum::bit::B0>, typenum::bit::B1>, typenum::bit::B1>, A> as std::convert::From<[A; 19]>>
          and 28 others
  = note: required by `std::convert::From::from`

error: aborting due to previous error

```

**Comment by pnkfelix** [comment]
This was injected between rustc 1.36.0-nightly (3991285f5 2019-04-25) and rustc 1.36.0-nightly (597f43248 2019-04-26).

**Comment by pnkfelix** [comment]
Ah... the code in question is this: 

https://github.com/bodil/sized-vec/blob/617bb45990c4ae734ebabd37f9746c4399db9554/src/lib.rs#L1082 

```rust
#[cfg(any(test, feature = "generic-array"))]
impl<N, A> From<generic_array::GenericArray<A, N>> for Vec<N, A>
where
    N: Unsigned + generic_array::ArrayLength<A>,
{
    /// Construct a vector of size `N` from a `GenericArray`.
    ///
    /// # Examples
    /// ```
    /// # #[macro_use] extern crate sized_vec;
    /// # use std::convert::TryFrom;
    /// # use sized_vec::Vec;
    /// # use typenum::U3;
    /// # use generic_array::GenericArray;
    /// # fn main() {
    /// let array = GenericArray::from([1, 2, 3]);
    /// let good_vec: Vec<U3, _> = Vec::from(array);
    /// assert_eq!(svec![1, 2, 3], good_vec);
    /// # }
    /// ```
...
```

Note in particular that this code is guarded by `cfg(any(test, feature = "generic-array"))`.

So, before  PR #59940 (which was in the rollup PR), we would not have tested that example code during `rustdoc --test`, because it would not have had `cfg(test)` set before that PR.

So presumably that code example fails to compile if one were to lift it to a higher point, outside of a doc example? I'll give that a quick whirl.

**Comment by jonas-schievink** [comment]
That PR might be getting reverted by https://github.com/rust-lang/rust/pull/61199 as well

**Comment by pnkfelix** [comment]
Well, in any case, when I lift the code in that test out of the docs, it fails to compile even on stable.

**Comment by pnkfelix** [comment]
(I still don't really understand how the logic for the cfg on that impl is meant to work. I would think that this `cfg` should just say `#[cfg(feature = "generic-array")]`; not `#[cfg(any(test, feature = "generic-array"))]`.)

**Comment by pnkfelix** [comment]
Filed https://github.com/bodil/sized-vec/issues/4 as a followup issue in the problematic crate.

Closing as not-a-bug, or at least not a regression that I think we need to act on specifically (though if we happen to "fix" it via #61199, I won't complain).
